lib/auth/native/native.go

```diff
@@ -51,34 +51,59 @@ type keyPair struct {
 
 // keygen is a key generator that precomputes keys to provide quick access to
 // public/private key pairs.
-type keygen struct {
+type Keygen struct {
 	keysCh chan keyPair
 
-	ctx    context.Context
-	cancel context.CancelFunc
+	ctx             context.Context
+	cancel          context.CancelFunc
+	precomputeCount int
+}
+
+// KeygenOption is a functional optional argument for key generator
+type KeygenOption func(k *Keygen) error
```

Comment:
Technically, the only KeygenOption you have now does not ever return an error, so you could get rid of the return parameter, and then you would not have to account for this error in all these places you had to change.